src/hooks/useLoginForm/useLoginForm.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect } from 'react';
+import { validationSchema } from './validations.js'
+
+
+const useLoginForm = (initialValues) => {
+  const [values, setValues] = useState(initialValues);
+  const [errors, setErrors] = useState(() => {
+    const tempErrors = {}
+    Object.keys(values).forEach(key => tempErrors[key] = null)
+    return tempErrors
+  })
+
+
+  // Private
+
+  const runValidations = (validations, value) => {
+    
+    return validations.map(validation => {
+      const isError = validation.validationFunction(value)
+      return isError ? validation.prompt : isError
+    })
+
+  }
```

Comment:
you can use implicit return
```suggestion
  const runValidations = (validations, value) => (
    validations.map(validation => {
      const isError = validation.validationFunction(value)
      return isError ? validation.prompt : isError
    })
  )
```

Reply:
yes, that's true